sdks/python/apache_beam/io/iobase.py

```diff
@@ -659,6 +675,167 @@ def fraction_consumed(self):
+    parallel. (3) an "unsplittable" input such as a cursor in a database.
+
+    consumed number of split points:
+
+      * Before a source read (``BoundedSource.read()`` invocation) claims the
+        first split point, number of consumed split points is 0. This condition
+        holds independent of whether the input is splittable.
+      * Any source read that has only claimed one split point for which
+        ``RangeTracker.set_done()`` has not been invoked has 0 consumed split
+        points: the first split point is the current split point and is still
+        being processed. This condition holds independent of whether the input
+        is splittable.
+      * For an empty source read (which never invokes
+        ``RangeTracker.try_claim()``), the consumed number of split points is 0.
+        This condition holds independent of whether the input is splittable.
+      * For a non-empty, finished source read (in which
+        ``RangeTracker.try_claim()`` was invoked at least once and
+        ``RangeTracker.set_done()`` has been invoked), the value returned must
+        be at least 1 and should equal the total number of split points in the
+        source.
+      * For example (1): After returning record #30 (starting at 1) out of 50 in
+        a perfectly splittable 50-record input, this value should be 29. When
+        finished, the number of split points consumed should be 50.
+      * For example (2): In a block-compressed value consisting of 5 blocks, the
+        value should stay at 0 until the first record of the second block is
+        returned; stay at 1 until the first record of the third block is
+        returned, etc. Only once the end-of-file is reached then the fifth block
+        has been consumed and the value should stay at 5.
+      * For example (3): For any non-empty unsplittable input, the consumed
+        number of split points is 0 until ``RangeTracker.set_done()`` is
+        invoked, at which point it becomes 1.
+
+    Remaining number of split points:
+
+    Assume for examples (1) and (2) that the number of records or blocks
+    remaining is known:
+
+    * Any source read for which ``RangeTracker.set_done()`` has not been
+      invoked, remaining number of split points should not be zero, because this
+      unfinished read itself represents a unfinished split point. This
+      condition holds independent of whether the input is splittable.
+    * A finished read, for which RangeTracker.set_done() has been invoked,
+      remaining number of split points is 0. This condition holds independent of
+      whether the input is splittable.
+    * For example 1: After returning record #30 (starting at 1) out of 50 in a
+      perfectly splittable 50-record input, remaining number of split points
+      should be 21 (20 remaining + 1 current) if the total number of records is
+      known.
+    * For example 2: After returning a record in block 3 in a block-compressed
```

Comment:
I think this section needs to be reworded in terms of blocks/records claimed, not records returned, as that (plus split requests) is the key time these numbers change. 

It might also be clearer to go through the examples one at a time, e.g. discussing the lifetime (consumed and remaining) for totally splittable sources first, then block sources, then unsplittable.